subsys/bluetooth/audio/ascs.c

```diff
@@ -1206,7 +1206,7 @@ static int ascs_ep_set_codec(struct bt_audio_ep *ep, uint8_t id, uint16_t cid,
 		.id = id,
 	};
 
-	if (ep == NULL && codec == NULL) {
+	if (ep == NULL) {
 		return -EINVAL;
```

Comment:
@Thalley so you can guarantee that if ep == NULL then codec also == NULL? Otherwise we can have NULL dereference.

Reply:
I assume you mean that if `ep != NULL` then `codec != NULL`? 

But yes, since `ascs_ep_set_codec` is only called one place, using the address of statically allocated values (which can not be NULL). In fact, in the current tree, `ep` can never be NULL.

Reply:
No I mean what I wrote. If ep == NULL and codec != NULL we have NULL dereference. If ep cannot be NULL please remove then NULL check.